src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -111,6 +112,15 @@ export const useSearch = (services: DiscoverViewServices) => {
     requests: new RequestAdapter(),
   };
 
+  const getDatasetAutoSearchOnPageLoadPreference = () => {
```

Comment:
I know it is a bit too much to ask. maybe we should add a test here given we have more and more logic in useSearch.

Reply:
+1 we definitely need to add a test for this